pontoon/base/migrations/0097_reject_suggestions_older_than_approved_translation.py

```diff
@@ -0,0 +1,44 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.10.7 on 2017-08-09 09:27
+from __future__ import unicode_literals
+
+from django.db import migrations
+
+
+"""
+Reject all suggestions that are older than the approved translation.
+
+Bug 1288956 added a `rejected` state to translations, and from then on, we consider
+everything that is neither approved nor rejected to be a suggestion. In order
+to reduce the amount of suggestions that will suddenly appear, we want to
+mark most of them as rejected.
+
+A translation will be rejected if:
+  * it is not approved, and
+  * it is older than either the submission date or the approval date
+    of the approved translation for its entity.
+"""
+SQL = """
+    UPDATE base_translation
+    SET rejected = TRUE
+    WHERE approved = FALSE
+    AND date < (
+        SELECT GREATEST(bt.date, bt.approved_date)
+        FROM base_translation bt
+        WHERE bt.approved = TRUE
+        AND bt.entity_id = base_translation.entity_id
+        AND bt.locale_id = base_translation.locale_id
+        ORDER BY bt.date, bt.approved_date
+        LIMIT 1
```

Comment:
Guess what? It seems we sometimes have several approved translations for an entity. At least, I've had an error with this sub query returning several results when running this on my local database. Anyway, this is a safe way to make sure we only take one.